src/renderers/dom/shared/ReactDOMComponent.js

```diff
@@ -270,6 +270,7 @@ if (__DEV__) {
       return;
     }
 
+    validateDOMNesting(null, String(content), this, this._ancestorInfo);
```

Comment:
Is this a good place to put it? Since the function is called `setContentChildForInstrumentation` I wouldn’t expect other logic to show up here, or maybe we can rename the function itself.